v2/robotmk/scheduler.py

```diff
@@ -197,6 +201,29 @@ def _run_attempts_until_successful(
             break
         return outputs
 
+    def _write_result_file(self, merged_xml_path: pathlib.Path) -> None:
```

Comment:
This is fine for me as a first version. However, I thought decided that moving files into the `results_directory` should be atomic?

Reply:
I had forgotten about that, you are right. Thanks!